server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -377,13 +377,21 @@ public interface IndexCommitListener {
          * {@link IndexCommitRef} prevents the {@link IndexCommitRef} files to be deleted from disk until the reference is closed. As such,
          * the listener must close the reference as soon as it is done with it.
          *
-         * @param shardId         the {@link ShardId} of shard
-         * @param store           the index shard store
-         * @param primaryTerm     the shard's primary term value
-         * @param indexCommitRef  a reference on the newly created index commit
-         * @param additionalFiles the set of filenames that are added by the new commit
+         * @param shardId             the {@link ShardId} of shard
+         * @param store               the index shard store
+         * @param primaryTerm         the shard's primary term value
+         * @param indexCommitRef      a reference on the newly created index commit
+         * @param additionalFiles     the set of filenames that are added by the new commit
+         * @param enginePreCommitData data extracted by the engine prior to the commit
          */
-        void onNewCommit(ShardId shardId, Store store, long primaryTerm, IndexCommitRef indexCommitRef, Set<String> additionalFiles);
+        void onNewCommit(
+            ShardId shardId,
+            Store store,
+            long primaryTerm,
+            IndexCommitRef indexCommitRef,
+            Set<String> additionalFiles,
+            Object enginePreCommitData
```

Comment:
now that I'm reading this again, I think that the name is misleading? this data was acquired _after_ the commit was created.

Reply:
(fyi "pre" I think meant "before the commit is exposed". but I see the slight confusion.)